### ISSUE TITLE
[BUG] Only remove() if pane still in DOM

### ISSUE THREAD

**Issue by rgoldiez** [roman-rr/cupertino-pane#138]
**Describe the bug**
We are looking to use this component to render a bottom panel on small screens.  When on larger screens, we will use a custom side panel.  When adjusting the window size to where the `cupertino-pane` gets removed and the side panel added, a `removeChild()` error is thrown because (I believe) this line: https://github.com/roman-rr/cupertino-pane/blob/ab846816426082a876db951f15b4bdb3f59dd15e/src/cupertino-pane.ts#L664

**To Reproduce**
1. Expose `cupertino-pane` in a react component that is toggled with a ternary operator.
2. While the pane is exposed, flip the logic on the ternary so the pane is no longer rendered.
3. See error

**Expected behavior**
Only `remove()` is child is still in the DOM.

**Additional context**
I'm running the latest version of `cupertino-pane`: 1.2.5

Let me know if you need a code sample.

**Comment by roman-rr** [comment]
@rgoldiez thank you for issue, i will investigate before next release.

**Comment by roman-rr** [comment]
Hello @rgoldiez let's try to find solution.

1. This line `this.wrapperEl.remove()` is called only with `.destroy()` process. 
2. In your case `.cupertino-pane-wrapper` element removed from DOM with some functions, and then additionally `.destroy()` called ? What if you call `.destroy()` before removed wrapper el from DOM ? 

Of course, we can just add statement `if (this.wrapperEl) {...}`, but better to think on logic.

**Comment by rgoldiez** [comment]
Hi @roman-rr - In our particular case, we're designing for mobile and web.  Based on screensize, we want to swap the `cupertino-pane` for a custom side panel (same content, but only one is rendered at a time).  In the case where `cupertino-pane` is showing and the screensize increases (browser window) we're switching from cupertino to the side panel - this is when we see the error.

The react component that abstracts bottom panel from side panel doesn't know about the inner workings of either panel - it just swaps between the two based on screen width.  Does this make sense?

**Comment by roman-rr** [comment]
@rgoldiez Please check if somewhere `.destroy()` method called in your code.

**Comment by rgoldiez** [comment]
@roman-rr - I use `destroy({animated: true})` in one place but the error even when commenting it out (and changing the window system to dynamically swap panels).

**Comment by roman-rr** [comment]
@rgoldiez Are you able to provide simple sources with this error ?

**Comment by rgoldiez** [comment]
@roman-rr - See if this helps.  Thank you!

https://codesandbox.io/s/cupertino-example-ny4wp?file=/src/App.tsx

**Comment by roman-rr** [comment]
@rgoldiez This issue ?

<img width="589" alt="Screen Shot 2021-03-13 at 02 26 11" src="https://user-images.githubusercontent.com/10646478/110988714-7f162e00-83a3-11eb-9b80-c1f98b59d2d5.png">

It is related to react, not to pane. 

Try to inset this snipped into bottom-panel.tsx (for debug)
```javascript
const { width } = useWindowSize();
const debouncedHandleResize = () => {
    if (width > 400) {
      destroy();
      let pane = document.querySelector(".cupertino-pane-wrapper");
      pane?.remove();
    }
};
window.addEventListener("resize", debouncedHandleResize);
```

Panel will be completely removed from DOM; But react still keep references somewhere and must be cleared also. 
Any suggestions ?

**Comment by rgoldiez** [comment]
@roman-rr - Thanks.  I dropped that snippet in and it dismisses (via destroy) at 400px width.  But at the threshold to switch from bottom to side, the `removeChild` issue resurfaces.

I was able to work around the issue by qualifying the "show" delivered to each panel with a boolean on which to display... and pulled out the ternary.  Example updated.  I'll close this issue.  Appreciate your help.

https://codesandbox.io/s/cupertino-example-ny4wp?file=/src/App.tsx
